compiler/rustc_trait_selection/src/solve/fulfill.rs

```diff
@@ -515,6 +515,30 @@ impl<'tcx> ProofTreeVisitor<'tcx> for BestObligation<'tcx> {
             self.with_derived_obligation(obligation, |this| nested_goal.visit_with(this))?;
         }
 
+        // alias-relate may fail because the lhs or rhs can't be normalized,
+        // and therefore is treated as rigid.
+        if let Some(ty::PredicateKind::AliasRelate(lhs, rhs, _)) = pred_kind.no_bound_vars() {
+            if let Some(obligation) = goal
+                .infcx()
+                .visit_proof_tree(
```

Comment:
This is jank and incorrect -- we're not inheriting a depth. Any ideas on a first-class way to create a custom nested obligation even though we don't have one in the proof tree? Should I just expose a constructor on `InspectGoal`?

Reply:
yeah, alternatively `visit_proof_tree_at_depth` or sth? Not sure, I don't 100% like the current proof tree API

Reply:
I'll do that for now. It shouldn't be difficult to change.